src/pytorch_lightning/trainer/connectors/data_connector.py

```diff
@@ -45,6 +45,10 @@
 
 warning_cache = WarningCache()
 
+BATCH_DATALOADER = Union[Collection[DataLoader], List[DataLoader], DataLoader[Any]]
+REQUEST_DATALOADER = Union[DataLoader, List[DataLoader], BATCH_DATALOADER]
+INSTANCE = Optional[Union[TRAIN_DATALOADERS, EVAL_DATALOADERS, "pl.LightningModule", "pl.LightningDataModule"]]
```

Comment:
Last nit: I'd rename this to something more descriptive

Reply:
it's only used once now; so I just moved the annotation back to the appropriate class attr in https://github.com/Lightning-AI/lightning/pull/13806/commits/c2431aeb555bace222bbe470d8a60c381458f125